PublicMapping/districtbuilder
# Share map page

## Overview

This PR makes a few different changes aimed at sharing Maps:
 - The map page can now be accessed for projects that don't belong to the user, or even when not logged in
 - There is a share link in the Map page header to copy the current page URL to the clipboard
 - UX changes were made to the map page for the new read-only mode:
   - The Accept button is hidden and the Cancel button is renamed to Clear
   - The block editing warning modal now appears every time you enter block editing mode, as we can't save to the server that you've already seen it as we do when you're logged in.
     We could store preferences like this in local storage for anonymous users as a follow-on task.
   - Similarly copy changes were made to the block editing modal to reflect that changes can't be persisted

### Checklist


### Demo

Sharing
![Peek 2020-09-25 08-12](https://user-images.githubusercontent.com/4432106/94266056-905ba300-ff07-11ea-9224-efa096cc2ea6.gif)

Read-only mode:
![Peek 2020-09-25 08-13](https://user-images.githubusercontent.com/4432106/94266068-96518400-ff07-11ea-9ccb-d226c8ea794c.gif)


### Notes

I thought about just never showing the block warning modal at all, and automatically clearing changes when switching geounits if you're in read-only mode, but figured we'd want to keep the behavior as close to editing-mode a possible.

## Testing Instructions

- `scripts/server
- Ensure the Map page continues to work normally when editing your own map
- Try going to a different users map while logged in, or going to a map while not logged in at all. It should work but in read-only mode
- Sharing from the link in the header should copy the page URL to the clipboard

Closes #126
